src/ucm/util/replace.c

```diff
@@ -23,8 +23,20 @@
 
 #define MAP_FAILED ((void*)-1)
 
-pthread_mutex_t ucm_reloc_get_orig_lock = PTHREAD_RECURSIVE_MUTEX_INITIALIZER_NP;
-pthread_t volatile ucm_reloc_get_orig_thread = -1;
+pthread_mutex_t ucm_reloc_get_orig_lock = PTHREAD_MUTEX_INITIALIZER;
+
+__attribute__((constructor(101)))
+void ucm_reloc_get_orig_lock_initializer()
+{
+    pthread_mutexattr_t mutattr = {0};
```

Comment:
why need to initialize it if pthread_mutexattr_init() is used?

Reply:
Probably don't. Will check.